vars/storybooks.groovy

```diff
@@ -0,0 +1,83 @@
+def getStorybooksBucket() {
+  return "ci-artifacts.kibana.dev/storybooks"
+}
+
+def getDestinationDir() {
+  return env.ghprbPullId ? "pr-${env.ghprbPullId}" : buildState.get('checkoutInfo').branch.replace("/", "__")
+}
+
+def getUrl() {
+  return "https://${getStorybooksBucket()}/${getDestinationDir()}"
+}
+
+def getUrlLatest() {
+  return "${getUrl()}/latest"
+}
+
+def getUrlForCommit() {
+  return "${getUrl()}/${buildState.get('checkoutInfo').commit}"
+}
+
+def upload() {
+  dir("built_assets/storybook") {
+    sh "mv ci_composite composite"
+
+    def storybooks = sh(
+      script: 'ls -1d */',
+      returnStdout: true
+    ).trim()
+      .split('\n')
+      .collect { it.replace('/', '') }
+      .findAll { it != 'composite' }
+
+    def listHtml = storybooks.collect { """<li><a href="${getUrlForCommit()}/${it}">${it}</a></li>""" }.join("\n")
```

Comment:
I'm a tiny bit concerned about creating HTML with string concatenation, though our sources are pretty predictable it would be great if we could limit the character set to `/[a-z\-_]/i` or something